07_Phyiscal Distributions fig draft

SimonTopp opened the pull request:
Contains code and figure for day 7, physical distributions. Could definitely fit in other categories too and am happy to move around if we'd prefer it to be elsewhere.  Also happy to get any feedback you have!

SimonTopp committed:
Started viz looking at summer lake color phenology.

SimonTopp committed:
Moved from animation to rayshader stack.

SimonTopp committed:
Added great lakes.

SimonTopp committed:
Fixed bug in fill colors

SimonTopp committed:
Added temporal distribution

SimonTopp committed:
Added readme and uploaded draft figure.

SimonTopp committed:
Added png

SimonTopp committed:
Masked out US for DEM

SimonTopp committed:
Updated to make aux plot elevation based.

SimonTopp committed:
Updated text and layout

SimonTopp committed:
added fig

SimonTopp committed:
Added doi and name

SimonTopp committed:
Removed xkcd text outside of the stickman

ceenell commented:
OK great, will look at this shortly. Can you write a couple sentences here about where the data came from and the takeaway message?

SimonTopp commented:
Totally, I included that stuff in the [07_physical_distributions_stopp/README.md](https://github.com/USGS-VIZLAB/chart-challenge-22/pull/8/files#diff-24398545fb65915856c7b0bbd65c6d1db465d918a2f7ccb439020f8184810ac8).  Copied below as well though:
- Overall messaging. How does the chart connect to the day/category? What is the 1-2 sentence takeaway?

Lake color is a key ecological indicator. Drivers of lake color, such as precipitation, ice duration, and land cover, are often correlated to lake elevation. High elevation lakes tend to have fewer nutrient inputs and limited seasonal light availability, leading to lower levels of phytoplankton growth and subsequently bluer colors. Only recently have we been able to examine continental patterns in key ecological indicators such as lake color. In doing so, we're able to better understand how these essential systems are changing in response to long term patterns in climate and land use practices.

- The data source and variables used. Where can the data be found? Is it from USGS or elsewhere? Did you do any pre-processing?

All data used can be downloaded with the provided code. Key datasets are 1) [LimnoSat-US](https://doi.org/10.5281/zenodo.4139694), which contains all cloud free Landsat observations of U.S. lakes from the HydroLakes database between 1984 and 2020 and 2) elevation data from USGS.

- Tools & libraries used

Stacked topography maps: rayshader and elevatr
Stick figure and annotation: xkcd
Layout: gridExtra, ggplot, and cowplot

lindsayplatt committed:
Merge branch 'main' of https://github.com/SimonTopp/chart-challenge-22 into SimonTopp-main

lindsayplatt committed:
commit an empty file so the folder needed exists!

lindsayplatt committed:
undoing because there is code to create the folder duh

lindsayplatt committed:
setwd to correct folder

lindsayplatt committed:
delete rogue URL

lindsayplatt commented on 07_physical_distributions_stopp/01_limnosat_download.R:
```diff
@@ -0,0 +1,47 @@
+}
+
+# Incerase out timeout because some of these are large files 
+options(timeout = max(6000, getOption("timeout")))
```
I have been attempting to download, but keep getting timeouts. I tried without setting a new timeout and with increasing this to 12,000 but haven't been able to successfully run any of the `download.file()` commands below. Any advice @SimonTopp ??

SimonTopp replied:
Unfortunately I didn't have the foresight to chunk up the dataset into smaller bits for easier downloads when I made it.  It's about ~3.7 gb and I've sporadically gotten stuck with timeouts when trying to download it programmatically.  If it's not working for you I'd suggest just downloading it from [zenodo directly](https://doi.org/10.5281/zenodo.4139694).

Edit: Just re-read and saw that you can't download any of them. I've only had issues with the big file (srCorrected_us_hydrolakes_dp_20200628.feather).  Are you using a mac or a windows? I'll go poke around a bit.

SimonTopp replied:
I think it'll be updated with a v2 in the next ~year or so, at which point we'll give a lot more thought to ease of use! For now it's a little unwieldy.

SimonTopp replied:
Just tried and everything works on my end (mac).  Could be you need to explicitly change the method to `libcurl`?

lindsayplatt replied:
Hmmm yeah, I am on Windows. Will change to `libcurl` and see what happens.

lindsayplatt replied:
Success!

lindsayplatt committed:
small edits to README

lindsayplatt committed:
small adjustment to wd in case someone just ran the download script

SimonTopp committed:
Removed random line in download.

lindsayplatt committed:
was able to use `curl` to download without any issues and without needing to change the timeout

lindsayplatt committed:
update gitignore

lindsayplatt commented on 07_physical_distributions_stopp/02_limnosat_viz.R:
```diff
@@ -0,0 +1,334 @@
+## Make our base layer
+elev_matrix %>%
+  sphere_shade(texture='desert') %>%
+  add_shadow(ray_shade(elev_matrix,zscale=50),0.3) %>%
```
Now stuck here. I get the lovely error

```
Error in hillshade[, , 1] * scales::rescale(shadowmap, c(max_darken, 1)) :                                                    
  non-conformable arrays
```

which [I have asked about before](https://github.com/tylermorganwall/rayshader/issues/46) and don't really understand in this context (all the matrices appear to be the same dimensions - 657x405)

SimonTopp replied:
What version `rayshader` are you using?  I think thats a bug that was fixed more recently than the issue you linked to (https://github.com/tylermorganwall/rayshader/issues/176) , but you may need to do devtools install from github and restart your R session.
`devtools::install_github("tylermorganwall/rayshader")`

lindsayplatt replied:
Good point! I installed `0.24.10` from CRAN today. I see `0.27.4` on github and will try with that version.

lindsayplatt replied:
Success with 0.27.4!

lindsayplatt commented:
FYI, I am working on being able to run this code from start to finish and will have a PR to your fork with some suggested changes to get it to run smoothly. In the meantime (darn you `hillshade ... non-conformable arrays` error), I thought I would share a few visual suggestions:

* In your matrix plot, I think it would be nice if the cells covered by ice are a light grey or something so it is clear they aren't just empty.
* I was thinking about how to translate this chart into a story on Twitter and that made me think that it would be nice if that map about lake counts was a separate chart from the rest. I think of it more as supporting information and I could see that + an bit about the lake count distribution would make a nice threaded tweet as part of the overall story.
* Could the matrix plot y-axis text label be horizontal and maybe at the top (right above 4000)? That will definitely make it more legible.
* I think it would be nice if it was clearer that this is summarizing data across over 35 years of data

Lastly, just want to confirm the following: 
* There are 42,473 lakes represented in this dataset (`length(unique(ls_elev$Hylak_id))`)
* To make the daily matrix plot, the lakes were grouped into 20 elevation bins representing 200 m each (`unique(daily_elev_color$elev_bin)`) and then the _median_ daily lake color was determined per bin (but only for bins where there was at least 50 lakes?)
* To make the lake color map, the lakes were grouped into a hex grid and the _median_ lake color for each hex cell for the full time period (March 1984 to June 2020) was calculated.

SimonTopp commented:
Thanks for the suggestions!  I'll address in more detail tomorrow, but a quick clarification:

>To make the daily matrix plot, the lakes were grouped into 20 elevation bins representing 200 m each (unique(daily_elev_color$elev_bin)) and then the median daily lake color was determined per bin (but only for bins where there was at least 50 lakes)

It's bins with at least 50 observations over the entire time span regardless of number of lakes.

lindsayplatt committed:
don't need this in the visualization script, so adding here

lindsayplatt committed:
needed a newer version than CRAN

lindsayplatt committed:
missing pipe symbol, errors otherwise

lindsayplatt committed:
was getting an error since 'x' didn't exist

lindsayplatt committed:
was getting an error since 'p_time' didn't exist. assume this was meant instead

lindsayplatt committed:
a bit more detail about the data download step

lindsayplatt committed:
delete lines that aren't used any more

lindsayplatt committed:
fix merge conflict (there was an extra blank line ...)

Merge branch 'main' of https://github.com/SimonTopp/chart-challenge-22 into SimonTopp-main

# Conflicts:
#	07_physical_distributions_stopp/01_limnosat_download.R

lindsayplatt commented:
OK, now that you have confirmed what is on the chart, I have a couple of text suggestions: 1) can the blue to green legend say `Median lake color over 36 years`, and 2) can the subtitle say `Distribution of over 42k lakes and their color for 36 years in relation to topography of the Contiguous U.S.`? Just to put a bit more context into it?

SimonTopp committed:
Addressed PR comments-moved y axis label, changed backround, adjusted title and legend text.

SimonTopp commented:
Ok, just addressed some of your suggestions:
- Moved y label to top of axis
- Adjust legend and title/subtitle text to include # of lakes and # of years
- Made a grey background for ice periods

I didn't move the count plot out because its more time consuming to adjust the layout and whatnot, one thought is if we want to make it a story in the tweet is to use the figure from the manuscript that has more info on the number of Landsat observations we have.
![Fig2_ObsDistribution](https://user-images.githubusercontent.com/33098707/162199833-a565badc-28a8-42cd-ab4a-140b5a49c8e2.png)

SimonTopp committed:
Moved annotation line a little.

SimonTopp committed:
Merge branch 'USGS-VIZLAB:main' into main

SimonTopp committed:
Updated gitignore

SimonTopp committed:
Merge branch 'pr/1'
